Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,447 @@
+    Assert::AreEqual(uint16_t(3000), metar.runway_visual_range_group.at(0).visibility_max);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.runway_visual_range_group.at(0).visibility_modifier);
+    metar.runway_visual_range_group.clear();
+
+    // R27L/2000FT
+    std::string m2("R27L/2000FT ");
+    metar::parsers::parse_runway_visual_range(metar, m2);
+    Assert::AreEqual(metar::runway_designator_type::left, metar.runway_visual_range_group.at(0).runway_designator);
+    Assert::AreEqual(uint8_t(27), metar.runway_visual_range_group.at(0).runway_number);
+    Assert::AreEqual(uint16_t(2000), metar.runway_visual_range_group.at(0).visibility_min);
+    Assert::AreEqual(uint16_t(2000), metar.runway_visual_range_group.at(0).visibility_max);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.runway_visual_range_group.at(0).visibility_modifier);
+    metar.runway_visual_range_group.clear();
+
+    // R06R/2000V3000FT
+    std::string m3("R06R/2000V3000FT ");
+    metar::parsers::parse_runway_visual_range(metar, m3);
+    Assert::AreEqual(metar::runway_designator_type::right, metar.runway_visual_range_group.at(0).runway_designator);
+    Assert::AreEqual(uint8_t(6), metar.runway_visual_range_group.at(0).runway_number);
+    Assert::AreEqual(uint16_t(2000), metar.runway_visual_range_group.at(0).visibility_min);
+    Assert::AreEqual(uint16_t(3000), metar.runway_visual_range_group.at(0).visibility_max);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.runway_visual_range_group.at(0).visibility_modifier);
+    metar.runway_visual_range_group.clear();
+
+    // R06R/2000V3000FT
+    std::string m4("R09L/3500V4500FT R09R/3000V4000FT ");
+    metar::parsers::parse_runway_visual_range(metar, m4);
+    Assert::AreEqual(metar::runway_designator_type::left, metar.runway_visual_range_group.at(0).runway_designator);
+    Assert::AreEqual(uint8_t(9), metar.runway_visual_range_group.at(0).runway_number);
+    Assert::AreEqual(uint16_t(3500), metar.runway_visual_range_group.at(0).visibility_min);
+    Assert::AreEqual(uint16_t(4500), metar.runway_visual_range_group.at(0).visibility_max);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.runway_visual_range_group.at(0).visibility_modifier);
+    Assert::AreEqual(metar::runway_designator_type::right, metar.runway_visual_range_group.at(1).runway_designator);
+    Assert::AreEqual(uint8_t(9), metar.runway_visual_range_group.at(1).runway_number);
+    Assert::AreEqual(uint16_t(3000), metar.runway_visual_range_group.at(1).visibility_min);
+    Assert::AreEqual(uint16_t(4000), metar.runway_visual_range_group.at(1).visibility_max);
+    Assert::AreEqual(metar::visibility_modifier_type::none, metar.runway_visual_range_group.at(1).visibility_modifier);
+    metar.runway_visual_range_group.clear();
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Weather()
```

Comment:
Can we add additional tests for the weather parser?  Here are some examples from Advisory Circular 00-45G (page 3-12):

Notation | Meaning
-------- | -------
-DZ | Light drizzle
-RASN | Light rain and snow
SN BR | (Moderate) snow, mist
-FZRA FG | Light freezing rain, fog
SHRA | (Moderate) rain shower
VCBLSA | Blowing sand in the vicinity
-RASN FG HZ | Light rain and snow, fog, haze
TS | Thunderstorm (without precipitation)
+TSRA | Thunderstorm, heavy rain
+FC TSRAGR BR | Tornado, thunderstorm, (moderate) rain, hail, mist